syntax = "proto2";

package tutorial;

message Person_State {
  required int32 player_id = 1;
  optional string player_name = 2;
  optional string current_weapon = 3;
  map<string, int32> weapon2bullet = 4;
  repeated float player_position = 5;
  optional int32 player_hp = 6;
  optional int32 player_armor = 7;

  enum Camp {
    BLUE = 0;
    RED = 1;
  }

  message Weapon{
    optional string Weapon_name = 1;
    optional int32 bullet = 2;
  }

 repeated Weapon weapons = 8;

}

message Gamer {
  repeated Person_State person = 1;
}
